backend/src/main/java/ai/verta/modeldb/entities/ProjectEntity.java

```diff
@@ -351,11 +354,16 @@ public Project getProtoObject(
       projectBuilder.setCodeVersionSnapshot(getCode_version_snapshot().getProtoObject());
     }
 
-    GetResourcesResponseItem projectResource =
-        roleService
-            .getEntityResource(
-                Optional.of(this.id), Optional.empty(), ModelDBServiceResourceTypes.PROJECT)
-            .get();
+    GetResourcesResponseItem projectResource;
+    if (getResourcesMap.isPresent() && !getResourcesMap.get().isEmpty()) {
+      projectResource = getResourcesMap.get().get(this.id);
```

Comment:
`projectResource` might still not be in the map. I'd rather be safer here just in case.

Reply:
Fixed